packages/extension-api/src/extension-api.d.ts

```diff
@@ -788,6 +789,93 @@ declare module '@podman-desktop/api' {
     connection: ContainerProviderConnection;
   }
 
+  /**
+   * Callback for openning shell session
+   */
+  export interface ProviderConnectionShellAccess {
+    /**
+     * Opens new session using ProviderConnectionShellAccessImpl class
+     * @example
+     * ```typescript
+     * const providerConnectionShellAccess = new ProviderConnectionShellAccessImpl(machineInfo);
+     * const session = providerConnectionShellAccess.open()
+     * ```
+     */
+    open(): ProviderConnectionShellAccessSession;
+  }
+
+  /**
+   * Callbacks for interaction with shell session
+   */
+  export interface ProviderConnectionShellAccessSession {
+    /**
+     * Receiving data event
+     * @example
+     * ```typescript
+     * session.onData(data => {...
+     * ```
+     */
+    onData: Event<ProviderConnectionShellAccessData>;
+
+    /**
+     * Error event
+     * @example
+     * ```typescript
+     * session.onError(error => {...
+     * ```
+     */
+    onError: Event<ProviderConnectionShellAccessError>;
+
+    /**
+     * End event
+     * @example
+     * ```typescript
+     * session.onEnd(onEnd);
+     * ```
+     */
+    onEnd: Event<void>;
+
+    /**
+     * Sends data
+     * @example
+     * ```typescript
+     * session.write(data)
+     * ```
+     */
+    write(data: string | Uint8Array): void;
+
+    /**
+     * Notifies server that terminal window has been resized
+     * @example
+     * ```typescript
+     * session.resize(dimensions)
```

Comment:
here I don't get as well the sample

it's not you that call resize, in the extension you implement the resize method